src/me/mrCookieSlime/Slimefun/Objects/MultiBlock.java

```diff
@@ -12,18 +13,56 @@
 public class MultiBlock {
 	
 	private Material[] blocks;
-	private Material trigger;
-	
+	private BlockFace trigger;
+	private boolean isSymmetric;
+
+	@Deprecated
 	public MultiBlock(Material[] build, Material trigger) {
+		this.blocks = build;
+		this.isSymmetric = isSymmetric(build);
+		//Hacky, the other constructor should be used
+		for (int i = 0; i < 9; i++)
```

Comment:
Iterating over every 9 elements is unnecessary if you only need element 1, 4, 7.
You should change the loop to 
for (int i = 1; i < 9; i *=3) to make it more performant.

Reply:
I have implemented it, but just wanted to mention three things: first this constructor shouldn't be used in the first place, second this is only called during bootup once per multiblock and third, as far as my knowledge goes, the difference between the old and new code is somewhere in the nanoseconds.
But anyways judging from my limited java bytecode and low machine code knowledge your solution should be faster.

Reply:
I never said it'd be a huge impact.
Empty statements are quite a bad-practice though, so you shouldn't really waste looping over items that will be ignored anyway, hence why I pointed it out.

Reply:
I agree with you on that one, bad coding practices should not be merged into the master branch